src/credential.rs

```diff
@@ -182,6 +183,146 @@ where
     }
 }
 
+pub struct BatchRequestBuilder<CR, JT, JE, JA>
+where
+    CR: CredentialRequestProfile,
+    JT: JsonWebKeyType,
+    JE: JweContentEncryptionAlgorithm<JT>,
+    JA: JweKeyManagementAlgorithm + Clone,
+{
+    body: BatchRequest<CR, JT, JE, JA>,
+    url: BatchCredentialUrl,
+    access_token: AccessToken,
+}
+
+impl<CR, JT, JE, JA> BatchRequestBuilder<CR, JT, JE, JA>
+where
+    CR: CredentialRequestProfile,
+    JT: JsonWebKeyType,
+    JE: JweContentEncryptionAlgorithm<JT>,
+    JA: JweKeyManagementAlgorithm + Clone,
+{
+    pub(crate) fn new(
+        body: BatchRequest<CR, JT, JE, JA>,
+        url: BatchCredentialUrl,
+        access_token: AccessToken,
+    ) -> Self {
+        Self {
+            body,
+            url,
+            access_token,
+        }
+    }
+
+    pub fn set_proofs<RE>(
+        mut self,
+        proofs_of_possession: Vec<Proof>,
+    ) -> Result<Self, RequestError<RE>>
+    where
+        RE: std::error::Error + 'static,
+    {
+        if self.body.credential_requests.len() != proofs_of_possession.len() {
+            return Err(RequestError::Other("invalid proof count".to_string()));
```

Comment:
If you could add the numbers to the error message it might make debugging easier if it ever occurs